libs/baf-contract/contract/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::env::keccak256;
+use near_sdk::AccountId;
+use near_sdk::{collections::UnorderedMap, env, near_bindgen};
+use ed25519_dalek::Verifier;
+use std::convert::TryInto;
+
+#[global_allocator]
+static ALLOC: near_sdk::wee_alloc::WeeAlloc = near_sdk::wee_alloc::WeeAlloc::INIT;
+
+type EdPK = [u8; 64];
+type SecpPK = [u8; 65];
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct AccountInfo {
+    ed_pk: EdPK,
+    accountId: AccountId,
+}
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct BafContract {
+    owner_id: AccountId,
+    account_infos: UnorderedMap<SecpPK, AccountInfo>,
+}
+
+#[near_bindgen]
+impl BafContract {
+    pub fn new() -> Self {
+        let owner_id = env::predecessor_account_id();
+        Self {
+            owner_id,
+            account_infos: UnorderedMap::new("account-infos-map".as_bytes()),
+        }
+    }
+
+    pub fn set_account_info(
```

Comment:
If this is being called from the frontend, need to add the following lines to the beginning of the function to the beginning to prevent spam and nonsense:

```rust
use near_sdk::env::{signer_account_id, is_valid_account_id, current_account_id};

if !is_valid_account_id(newAccoundID.as_bytes()) {
    return Err("newAccountID is invalid!");
}

let signer = signer_account_id();
if signer != newAccountID && signer != current_account_id() {
    return Err("signer must own either newAccountID or the contract itself!");
}
```